src/Types/Workout.ts

```diff
@@ -0,0 +1,114 @@
+import { Resources } from './Resources';
+export type WorkoutObjectType =
+  | 'exuipment'
+  | 'position'
+  | 'exercise'
+  | 'assignment'
+  | 'set'
+  | 'workout';
+
+export type Reference<T> = string | T;
+
+export interface WorkoutObject {
+  name: string; // unique string used to reference the object
+  label?: string; // string used for display, defaults to name
+  type?: WorkoutObjectType; // if not speified will be inffered
```

Comment:
Inferring the type is difficult, especially with how these can evolve in the future. I suggest not to allow this to be `null`